src/en/space-station-14/core-tech/proposals/appearancesystem-compliance.md

```diff
@@ -0,0 +1,108 @@
+
+### Proposed Solution, implementation
+
+To achieve the overview, the following implementation would be required:
+
+- Any game logic system that modifies sprites is split up into a game logic system and a visualizer system (i.e. inherits `VisualizerSystem`). The game logic system *sets* `AppearanceData` for the visualizer system to work with, instead of modifying the sprites directly.
+- All visualizer systems are connected to visualizer components (e.g. `ClothingSystem` have `ClothingVisualizerSystem` and `ClothingVisualizerComponent`). The visualizer system handles the visual logic, and the component on an entity indicates an entity should run the system.
+- `SpriteSystem`/`SpriteComponent` may not be accessed outside of visualizer systems to prevent game logic systems modifying sprites.
+- Visualizer systems must be connected to a visualizer component (via inheritance, `[ExampleVisualizerSystem] : VisualizerSystem<[ExampleVisualizerComponent]>`; this is enforced with a new Attribute tag `[VisualizerComponent]` that ensures the component is only tied to a system that inherits the appropriate `VisualizerSystem`.
+  - This tag is also what allows easy copying of visualizers between objects, as the copying entity can check for all components with the tag.
+ 
+Entities are free to modify the `AppearanceData` of other entities. `AppearanceData` should not be used to store game logic and should be kept as lightweight as possible; only data used to modify the entity's base prototype sprite should be included.
+
+The use of visualizer systems and components, as well as access to `SpriteSystem`, can be mechanically enforced in the codebase. There would need to be attention paid during code review to ensure there is no game logic performed inside of visualizer systems.
+ 
+### Issues
+
+This implementation does have its downsides:
+- You end up with a lot more systems and components!
+  - This could be mitigated with attention to file sorting and multiple game logic system sharing a single visualizer, similar to how [WieldableSystem](https://github.com/space-wizards/space-station-14/blob/10877ebbf975ac2c75a13f3b6ef146166acce049/Content.Shared/Wieldable/WieldableSystem.cs) handles all game logic components adjacent to wielding.
+- Ensuring visualizer system order isn't clear!
+  - This could possibly be mitigated with the use of the `after:` keyword when subscribing to the `OnAppearanceDataChanged` event, so that for example the rotation caused by lying down overwrites the rotation caused by waddling.
+
+### Animations
+
+Animations (especially looping ones) would require syncing the start time/offset of the animation frame, so that if a new client loads the animating entity it can accurately set the sprite's animation. This needs to be replicated across multiple systems such as lights and movement as well. 
+
+### Examples of concrete uses
+
+- Remembering sprites at point in time
+  - End of round sprites
+  - Photography
```

Comment:
TBH I'm somewhat tempted to just move sprite component back to shared (with networking disabled by default) to make server-authoritative sprites & taking sprite snapshots much easier. Though that'd also have a performance impact on the server, though I have NFI if it'd be significant. It'd also be a giant refactor that'd require moving all visualizers to shared.

And while maybe somewhat overkill, sprite snapshots could also be obtained by actually serializing the whole entity & spawning a client-side dummy for rendering.